src/asm_ghidra.cpp

```diff
@@ -30,6 +38,8 @@ static int disassemble(RAsm *a, RAsmOp *op, const ut8 *buf, int len)
 		r = 1;
 	}
 #endif
+	if(rio)
+		r_io_free(rio);
```

Comment:
the if (notnull)free(notnull) is a bad nattern

Reply:
Change to what?

Reply:
Just `r_io_free(rio)` without the if. `free(NULL)` in C is valid and r2's free functions follow this pattern too.

Reply:
get it.

Reply:
you are not the owner of rio in bin.